src/legacy/ui/public/saved_objects/show_saved_object_save_modal.tsx

```diff
@@ -30,21 +44,18 @@ export function showSaveModal(saveModal) {
 
   const onSave = saveModal.props.onSave;
 
-  const onSaveConfirmed = (...args) => {
-    onSave(...args).then(({ id, error }) => {
-      if (id || error) {
-        closeModal();
-      }
-    });
+  const onSaveConfirmed: MinimalSaveModalProps['onSave'] = async (...args) => {
+    const response = await onSave(...args);
+    if (('id' in response && response.id) || ('error' in response && response.error)) {
```

Comment:
The `'id' in response` is necessary because it acts as a type guard to make sure it's the right kind of reponse. I agree that it's a bit verbose, but we gain type safety by doing that. Putting the type guard in a helper function makes this more readable, I can do that.

Reply:
Not critical, but nice to have

Reply:
It's in the newest version, plus a comment where it's used